test/e2e/dlb/dlb.go

```diff
@@ -0,0 +1,98 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dlb
+
+import (
+	"context"
+	"path/filepath"
+	"time"
+
+	"github.com/intel/intel-device-plugins-for-kubernetes/test/e2e/utils"
+	"github.com/onsi/ginkgo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/framework/kubectl"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+)
+
+const (
+	kustomizationYaml = "deployments/dlb_plugin/kustomization.yaml"
+)
+
+func init() {
+	ginkgo.Describe("dlb plugin", describe)
+}
+
+func describe() {
+	f := framework.NewDefaultFramework("dlbplugin")
+
+	kustomizationPath, err := utils.LocateRepoFile(kustomizationYaml)
+	if err != nil {
+		framework.Failf("unable to locate %q: %v", kustomizationYaml, err)
+	}
+
+	ginkgo.It("checks availability of DLB resources", func() {
+		ginkgo.By("deploying DLB plugin")
+		framework.RunKubectlOrDie(f.Namespace.Name, "--namespace", f.Namespace.Name, "apply", "-k", filepath.Dir(kustomizationPath))
+
+		ginkgo.By("waiting for DLB plugin's availability")
+		if _, err := e2epod.WaitForPodsWithLabelRunningReady(f.ClientSet, f.Namespace.Name,
+			labels.Set{"app": "intel-dlb-plugin"}.AsSelector(), 1 /* one replica */, 100*time.Second); err != nil {
+			framework.DumpAllNamespaceInfo(f.ClientSet, f.Namespace.Name)
+			kubectl.LogFailedContainers(f.ClientSet, f.Namespace.Name, framework.Logf)
+			framework.Failf("unable to wait for all pods to be running and ready: %v", err)
+		}
+
+		ginkgo.By("checking the resource is allocatable")
+		if err := utils.WaitForNodesWithResource(f.ClientSet, "dlb.intel.com/pf", 30*time.Second); err != nil {
+			framework.Failf("unable to wait for nodes to have positive allocatable resource: %v", err)
+		}
+		ginkgo.By("checking the resource is allocatable")
+		if err := utils.WaitForNodesWithResource(f.ClientSet, "dlb.intel.com/vf", 30*time.Second); err != nil {
+			framework.Failf("unable to wait for nodes to have positive allocatable resource: %v", err)
+		}
+
+		ginkgo.By("submitting a pod requesting DLB pf resources")
+		podSpec := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{Name: "dlbplugin-tester"},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Args:    []string{"-c", "echo hello world"},
+						Name:    "testcontainer",
+						Image:   imageutils.GetE2EImage(imageutils.BusyBox),
+						Command: []string{"/bin/sh"},
+						Resources: v1.ResourceRequirements{
+							Requests: v1.ResourceList{"dlb.intel.com/pf": resource.MustParse("1"),
+								"dlb.intel.com/vf": resource.MustParse("1")},
```

Comment:
todo: test with vf and pf resources separately.

Reply:
Ah, was what Ed said that it would be better to just separate into 2 pods? I tested with 2 containers in one pod (that is the same as demo yaml file) with various conditions. And I think this actually does not harm the "easiness" to recognize where the problem happened even if they are in one pod.! If it's still better to divide, tell me. Or, if I misunderstood, or remember wrongly, tell me.!

Reply:
I said that would be better to have two separate test cases for pf and vf, but I'm not insisting. If it's easy to understand which part of test breaks (pf or vf) I'm ok with the current implementation.

Reply:
@bart0sh Ok, thanks, then,,,
I am not insisting either,, but can we just keep it as it is now? :)

Reply:
@mythi wdyt?

Reply:
"as it is now" is not very good because the test says it checks a pod with PF resource requests but ends up requesting both PF and VF.

Should be easy to instantiate one `podSpec` and set/change Requests/Limits or both VF and PF (maybe later with different images) too.

Reply:
> "as it is now" is not very good because the test says it checks a pod with PF resource requests but ends up requesting both PF and VF.

Ah!! Why did I put there pf? I changed.!